pkg/controllers/runnable_reconciler_test.go

```diff
@@ -45,10 +53,252 @@ import (
+		Repo:                    controllerRepo,
+		DependencyTracker:       dependencyTracker,
+		ConditionManagerBuilder: conditions.NewConditionManager,
+	}
+}
+
+// These tests are still in flux. So grouping by context is still up in the air
+// Todo: need to test present service account by default and by param
+func TestMissingServiceAccount(t *testing.T) {
+	runnableNamespace := "test-ns"
+	runnableName := "my-runnable"
+	runnableRequest := controllerruntime.Request{NamespacedName: types.NamespacedName{Namespace: runnableNamespace, Name: runnableName}}
+
+	serviceAccountName := "my-service-account"
+
+	now := metav1.Now()
+
+	baseServiceAccount := v1.ServiceAccountBlank.
+		MetadataDie(func(d *diesv1.ObjectMetaDie) {
+			d.Namespace(runnableNamespace)
+			d.Name(serviceAccountName)
+			d.CreationTimestamp(now)
+		})
+
+	baseRunnable := dies.RunnableBlank.
+		MetadataDie(func(d *diesv1.ObjectMetaDie) {
+			d.
+				Name(runnableName).
+				Namespace(runnableNamespace).
+				AddLabel("some-val", "first")
+		}).
+		SpecDie(func(d *dies.RunnableSpecDie) {
+			d.
+				RetentionPolicy(v1alpha1.RetentionPolicy{
+					MaxFailedRuns:     10,
+					MaxSuccessfulRuns: 10,
+				}).
+				RunTemplateRef(v1alpha1.TemplateReference{
+					Kind: "ClusterRunTemplate",
+					Name: "my-run-template",
+				})
+		})
+
+	secretName := "my-secret"
+
+	scheme := runtime.NewScheme()
+	_ = clientgoscheme.AddToScheme(scheme)
+	_ = v1alpha1.AddToScheme(scheme)
+
+	rts := rtesting.ReconcilerTests{
+		"default service account missing": {
+			Request:           runnableRequest,
+			AdditionalConfigs: nil,
+			GivenObjects: []client.Object{
+				baseRunnable,
+			},
+			ExpectStatusUpdates: []client.Object{
+				baseRunnable.
+					StatusDie(func(d *dies.RunnableStatusDie) {
+						d.ConditionsDie(
+							diesv1.ConditionBlank.
+								Type("RunTemplateReady").
+								Status(metav1.ConditionFalse).
+								Reason("ServiceAccountSecretError").
+								Message(`failed to get service account object from api server [test-ns/default]: failed to get object [test-ns/default] from api server: serviceaccounts "default" not found`),
+							diesv1.ConditionBlank.
+								Type("Ready").
+								Status(metav1.ConditionFalse).
+								Reason("ServiceAccountSecretError").
+								Message(`failed to get service account object from api server [test-ns/default]: failed to get object [test-ns/default] from api server: serviceaccounts "default" not found`),
+						)
+					}),
+			},
+		},
+		"provided service account missing": {
+			Request:           runnableRequest,
+			AdditionalConfigs: nil,
+			GivenObjects: []client.Object{
+				baseRunnable.SpecDie(func(d *dies.RunnableSpecDie) {
+					d.ServiceAccountName("my-service-account")
+				}),
+			},
+			ExpectStatusUpdates: []client.Object{
+				baseRunnable.
+					StatusDie(func(d *dies.RunnableStatusDie) {
+						d.ConditionsDie(
+							diesv1.ConditionBlank.
+								Type("RunTemplateReady").
```

Comment:
**suggestion** with dies, if you find yourself repeating the same pattern, it can be helpful to store the common bits. Common conditions while structurally generic, are semantically part of the type domain. Along side the other die types you can define something like.

```go
var RunTemplateReadyCondition := diesv1.ConditionBlank.Type("RunTemplateReady")
```

and then consume it

```suggestion
							dies.RunTemplateReadyCondition.
```

Reply:
Yup I saw that in other examples, and don't personally like it. Perhaps after driving the tests out a few more times I'd appreciate the encapsulation and abstraction, but when driving out code with tests, I don't like to hide meaning behind indirection, at least not early. I want to see at a glance what's noticeably different. I know plenty of folks skew the other way.